exiftool_test.go

```diff
@@ -357,5 +361,369 @@ func TestSetExiftoolBinaryPath(t *testing.T) {
 	// error on init
 	_, err = NewExiftool(SetExiftoolBinaryPath("/non/existing/path"))
 	assert.NotNil(t, err)
+}
+
+func TestWriteMetadataSuccessTokenHandling(t *testing.T) {
+	testCases := []struct{
+		name string
+		testResp string
+		expectErr bool
+	}{
+		{name: "token as full resp", testResp: writeMetadataSuccessToken, expectErr: false},
+		{name: "token at resp end", testResp: "prefix text" + writeMetadataSuccessToken,
+			expectErr: false},
+		{name: "token at resp middle",
+			testResp: "prefix text" + writeMetadataSuccessToken + "suffix text",
+			expectErr: true},
+		{name: "token at resp beginning", testResp: writeMetadataSuccessToken + "suffix text",
+			expectErr: true},
+		{name: "no token", testResp: "some error message", expectErr: true},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T){
+			err := handleWriteMetadataResponse(tc.testResp)
+			if tc.expectErr {
+				assert.Error(t, err)
+			} else {
+				assert.Nil(t, err)
+			}
+		})
+	}
+}
+
+var fieldsForWriting = map[string]interface{} {
+	"Title": "fake title",
+	"Description": "fake description",
+	// TODO: Test ints
+	//       Not testing ints since we'd need to update the
+	//       decoder and FileMetadata to use UseNumber()
+	// "CameraImagingModelImageHeight": 5, // test ints
```

Comment:
I don't understand

Reply:
There's another test case that should be implemented : the array.

Reply:
Here is a modified version of `testdata/20190404_131804.jpg`, enriched with keywords:

```
barasher@linux:~/Downloads$ exiftool 20190404_131804.jpg -j | grep Keywords
  "Keywords": ["keyword with space","keyword1","keyword2"],
```

![20190404_131804](https://user-images.githubusercontent.com/4041420/126701936-82a6ab93-592a-417c-a37b-90a0fcdf8f7a.jpg)

Reply:
Tests pass now. I'm guessing it fails with older versions of exiftool. Determining compatibility with different versions of exiftool should be out of the scope of this PR. FWIW, I tested against version 12.26

I don't have a use case for supporting arrays yet and don't have much more bandwidth to iterate on this PR. Is supporting arrays a blocker for merging this PR? We can always add tests for this later. I explicitly avoided `+=` and `-=` operations for tags and only supported `=` for simplicity.

Reply:
> I don't understand

The JSON unmarshaller/decoder interprets all numerics as float64 by default so you can't test integers using `json.Unmarshal()` e.g. you'd need to use `json.Decoder` with `UseNumber()` to test int tag values

Docs:
* https://pkg.go.dev/encoding/json#Decoder.UseNumber

Reply:
> > I don't understand
> 
> The JSON unmarshaller/decoder interprets all numerics as float64 by default so you can't test integers using `json.Unmarshal()` e.g. you'd need to use `json.Decoder` with `UseNumber()` to test int tag values
> 
> Docs:
> 
> * https://pkg.go.dev/encoding/json#Decoder.UseNumber

I didn't know, thanks for the explanation

Reply:
> Tests pass now. I'm guessing it fails with older versions of exiftool. Determining compatibility with different versions of exiftool should be out of the scope of this PR. FWIW, I tested against version 12.26
> 
> I don't have a use case for supporting arrays yet and don't have much more bandwidth to iterate on this PR. Is supporting arrays a blocker for merging this PR? We can always add tests for this later. I explicitly avoided `+=` and `-=` operations for tags and only supported `=` for simplicity.

Okay, deal ! Let's do it later.